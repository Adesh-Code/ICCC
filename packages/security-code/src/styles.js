const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
