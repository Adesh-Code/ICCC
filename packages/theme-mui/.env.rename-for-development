# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE