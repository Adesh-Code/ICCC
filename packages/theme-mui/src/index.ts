// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
