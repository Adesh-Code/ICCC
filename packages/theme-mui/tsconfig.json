{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
