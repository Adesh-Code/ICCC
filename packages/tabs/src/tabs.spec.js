import React from "react";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import Tab from "@mui/material/Tab";
import Tabs from "./tabs";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

const arrange = ({ children = <Tab label="label" />, ...otherProps } = {}) => {
  const utils = render(
    <ThemeProvider theme={light}>
      <Tabs {...otherProps}>{children}</Tabs>
    </ThemeProvider>
  );
  return utils;
};

describe("Tabs component", () => {
  beforeEach(() => {
    // eslint-disable-next-line no-console
    console.error = jest.fn((error) => {
      throw new Error(error);
    });
  });

  it("renders tabs", () => {
    const { container } = arrange();
    expect(container).toMatchSnapshot();
  });

  it("does not render if no children are provided", () => {
    expect(() => arrange({ children: null })).toThrow();
  });

  it("accepts classnames", () => {
    const { container } = arrange({
      className: "foo"
    });
    const tabs = container.firstChild;
    expect(tabs).toHaveClass("foo");
  });

  it("renders the default style variant", () => {
    const { container } = arrange();
    const tabs = container.firstChild;
    expect(tabs).toHaveClass("DhigTabs--styleVariant--default");
  });

  it("renders the quiet style variant", () => {
    const { container } = arrange({ styleVariant: "quiet" });
    const tabs = container.firstChild;
    expect(tabs).toHaveClass("DhigTabs--styleVariant--quiet");
  });

  it("renders the quiet style variant with the underline extended", () => {
    const { container } = arrange({
      styleVariant: "quiet",
      extendedUnderline: true
    });
    const tabs = container.firstChild;
    expect(tabs).toHaveClass("DhigTabs--extendedUnderline");
  });

  it("chooses the correct tab on click", async () => {
    arrange({
      children: [
        <Tab key={0} value={0} label="Tab 0" />,
        <Tab key={1} value={1} label="Tab 1" />,
        <Tab key={2} value={2} label="Tab 2" />
      ],
      value: 1
    });

    const tab1 = screen.getByText("Tab 1");
    expect(tab1).toHaveAttribute("aria-selected", "true");

    const tab2 = screen.getByText("Tab 2");
    fireEvent.click(tab2);

    await waitFor(() => {
      expect(tab2).toHaveAttribute("aria-selected", "true");
    });
  });

  it("accepts custom onChange", async () => {
    let value = 0;
    arrange({
      children: [
        <Tab key={0} value={0} label="Tab 0" />,
        <Tab key={1} value={1} label="Tab 1" />,
        <Tab key={2} value={2} label="Tab 2" />
      ],
      value,
      onChange: (event) => (value = event.target.value)
    });
    const tab2 = screen.getByText("Tab 2");
    fireEvent.click(tab2);
    await waitFor(() => {
      expect(tab2).toHaveAttribute("aria-selected", "true");
    });
  });

  it("renders with accessibility props", () => {
    arrange({
      "aria-label": "aria-label-test"
    });
    const tabs = screen.getByRole("tablist");
    expect(tabs).toHaveAttribute("aria-label", "aria-label-test");
  });

  it("applies the correct styles to the selected tab", () => {
    arrange();
    const tab = screen.getByRole("tab");
    fireEvent.click(tab);
    expect(tab).toHaveClass("Mui-selected");
  });
});
