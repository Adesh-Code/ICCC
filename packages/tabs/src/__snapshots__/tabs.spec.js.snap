// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component renders tabs 1`] = `
<div>
  <div
    class="MuiTabs-root DhigTabs--styleVariant--default emotion-0"
  >
    <div
      class="MuiTabs-scrollableX MuiTabs-hideScrollbar emotion-1"
      style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
    />
    <div
      class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX emotion-2"
      style="margin-bottom: 0px;"
    >
      <div
        class="MuiTabs-flexContainer emotion-3"
        role="tablist"
      >
        <button
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-wrapped Mui-selected emotion-4"
          role="tab"
          tabindex="0"
          type="button"
        >
          label
        </button>
      </div>
      <span
        class="MuiTabs-indicator emotion-5"
        style="left: 0px; width: 0px;"
      />
    </div>
  </div>
</div>
`;
